dek-registry-client/src/main/java/io/confluent/dekregistry/client/CachedDekRegistryClient.java

```diff
@@ -284,11 +284,15 @@ public Dek createDek(
     request.setAlgorithm(algorithm);
     request.setEncryptedKeyMaterial(encryptedKeyMaterial);
     request.setDeleted(deleted);
-    Dek dek = restService.createDek(requestProperties, kekName, request);
-    dekCache.put(new DekId(kekName, subject, version, algorithm, deleted), dek);
-    dekCache.invalidate(new DekId(kekName, subject, LATEST_VERSION, algorithm, false));
-    dekCache.invalidate(new DekId(kekName, subject, LATEST_VERSION, algorithm, true));
-    return dek;
+    try {
+      Dek dek = restService.createDek(requestProperties, kekName, request);
+      dekCache.put(new DekId(kekName, subject, version, algorithm, deleted), dek);
+      return dek;
+    } finally {
+      // Ensure latest dek is invalidated, such as in case of conflict (409)
```

Comment:
Is this not required for Delete and UnDelete?

Reply:
I don't believe so